### PR TITLE
[DO NOT MERGE] Add new kubeVersion check and annotation

### PR DESCRIPTION
**Pull request by mmulholla**
DO NOT MERGE, merge needs to sync with: https://github.com/openshift-helm-charts/development/pull/59

Adds a new ```v1.1/has-kubeversion``` check to the 1.1 profiles.

It ensure sthe chart kubeVersion is set to a valid semantic version. From this the supported OpenShiftVersions are determined and included in an annotation in the report.  

Also added to profile 1.1 the rename  of annotation : certifiedOpenShiftVersions to testedOpenShiftVersion

### CODE REVIEW

File: pkg/chartverifier/checks/checks.go [hunk 1]
```diff
@@ -337,3 +361,40 @@ func parseImageReference(image string) pyxis.ImageReference {
...
+
+func getOCPRange(kubeVersionRange string) string {
+
+	semverCompare := sprig.GenericFuncMap()["semverCompare"].(func(string, string) (bool, error))
+	minOCPVersion := ""
+	maxOCPVersion := ""
+	for kubeVersion, OCPVersion := range tool.GetKubeOpenShiftVersionMap() {
+		match, err := semverCompare(kubeVersionRange, kubeVersion)
+		if err != nil {
+			return fmt.Sprintf("%s : %s", KuberVersionProcessingError, err)
```
Comment (zonggen) [inline pkg/chartverifier/checks/checks.go:373]: Nit: can we use the common style for error handling in golang: https://golang.org/doc/effective_go#multiple-returns? i.e. return result as well as the error if any (`func getOCPRange(kubeVersionRange string) string error`).

File: pkg/chartverifier/checks/checks.go [hunk 2]
```diff
@@ -337,3 +361,40 @@ func parseImageReference(image string) pyxis.ImageReference {
...
+			if minOCPVersion == "" || semver.Compare(testOCPVersion, fmt.Sprintf("v%s", minOCPVersion)) < 0 {
+				minOCPVersion = OCPVersion
+			}
+			if maxOCPVersion == "" || semver.Compare(testOCPVersion, fmt.Sprintf("v%s", maxOCPVersion)) > 0 {
+				maxOCPVersion = OCPVersion
+			}
+		}
+	}
+	if minOCPVersion != "" {
+		match, _ := semverCompare(kubeVersionRange, "1.999")
```
Comment (zonggen) [inline pkg/chartverifier/checks/checks.go:386]: Confused in this condition check, why are we comparing to 1.999 here?

File: pkg/chartverifier/report/testreports/v1.0/reportallsorts.yaml [hunk 3]
```diff
@@ -11,7 +11,7 @@ metadata:
         digests:
             chart: sha256:0c1c44def5c5de45212d90396062e18e0311b07789f477268fbf233c1783dbd0
         lastCertifiedTimestamp: "2021-07-02T08:09:56.881793-04:00"
-        certifiedOpenShiftVersions: N/A
+        certifiedOpenShiftVersions : N/A
```
Comment (zonggen) [inline pkg/chartverifier/report/testreports/v1.0/reportallsorts.yaml:14]:
```suggestion
        certifiedOpenShiftVersions: N/A
```

File: pkg/chartverifier/checks/checks.go (hunk 2 above)
Comment (zonggen) [reply pkg/chartverifier/checks/checks.go:386]: Got it. Thanks for explaining! I think a comment as a note will be helpful.

File: pkg/chartverifier/checks/checks.go (hunk 1 above)
Comment (mmulholla) [reply pkg/chartverifier/checks/checks.go]: Fixed

File: pkg/chartverifier/report/testreports/v1.0/reportallsorts.yaml (hunk 3 above)
Comment (mmulholla) [reply pkg/chartverifier/report/testreports/v1.0/reportallsorts.yaml]: Fixed

File: pkg/chartverifier/checks/checks.go (hunk 2 above)
Comment (mmulholla) [reply pkg/chartverifier/checks/checks.go:388]: Comment now added.

**Review by dperaza4dustbit — APPROVED** [review]
lgtm

**Review by zonggen — APPROVED** [review]
LGTM!

**Pull request merged**